tests/test_formatter.py

```diff
@@ -203,3 +203,87 @@ def test_escape(self):
 
         # Escaping is atomic: brackets inside brackets are treated literally
         assert self.formatter.format(datetime(1, 1, 1), "[[[ ]]") == "[[ ]"
+
+
+class TestDateTimeFormatterPremadeFormats:
+    def setup_class(cls):
+        cls.formatter = formatter.DateTimeFormatter()
+
+    # cookie format: 'Thursday, 25-Dec-1975 14:15:16 EST'
+    def test_cookie(self):
+        tz = list(make_full_tz_list())[0]
+        dt = datetime(1975, 12, 25, 14, 15, 16, tzinfo=dateutil_tz.gettz(tz))
+
+        correct_timezone = self.formatter._format_token(dt, "ZZZ")
+        assert (
+            self.formatter.format_cookie(dt)
+            == "Thursday, 25-Dec-1975 14:15:16 " + correct_timezone
+        )
+
+        # Testing that single digit values return with a 0 in front
+        dt_single_digit = datetime(
+            1975, 2, 5, 4, 5, 6, tzinfo=pytz.timezone("UTC")
+        ).replace(tzinfo=dateutil_tz.gettz(tz))
```

Comment:
```suggestion
            1975, 2, 5, 4, 5, 6, tzinfo=dateutil_tz.gettz(tz)
        )
```
No need to replace, we can start with the right tz straight away.